### PR TITLE
feat(native): Set sdk.name for events created from minidumps

### LINKED ISSUE #4236: Add `sdk.name` to events created out of minidumps
A lot of diagnostics in Sentry are done by looking up `sdk.name`.
Some of our telemetry internally is also based on that, so we better understand volume of crashes by technology for example.

For example, on Relay we add `unreal.crashreporter` to events from the unreal endpoint:

https://github.com/getsentry/relay/blob/d12ff531fefca0b1c0dbe2b4efb72032abed74d2/relay-server/src/utils/unreal.rs#L25

Suggested SDK name: `minidump`.
Since it's not "Sentry" specific, we I imagine we don't know what tool created it.

If we know it's breakpad vs crashpad vs other, i'd suggest: `minidump.crashpad`, `minidump.breakpad`, `minidump.dotnet` (dotnet has its own minidump creation for example with support for it's JIT information).

### PR DESCRIPTION
**Pull request by tobias-wilfert**
Sets the sdk.name for events created out of minidumps to `minidump`, mimicking the logic that we currently have for [unreal](https://github.com/getsentry/relay/blob/d12ff531fefca0b1c0dbe2b4efb72032abed74d2/relay-server/src/utils/unreal.rs#L281).

Not sure we can add more detail than just `minidump`? Presumably something like this could work but I am not sure if we want to do that:
```rs
  let more_detailed_sdk_name = if minidump.get_stream::<MinidumpCrashpadInfo>().is_ok() {
      "minidump.crashpad"
  } else if minidump.get_stream::<minidump::MinidumpBreakpadInfo>().is_ok() {
      "minidump.breakpad"
  } else {
      "minidump.unknown"
  };
```

Fix: https://github.com/getsentry/relay/issues/4236

### CODE REVIEW

File: relay-server/src/utils/native.rs [hunk 1]
```diff
@@ -9,12 +9,15 @@ use std::error::Error;
...
 use relay_event_schema::protocol::{
-    Context, Contexts, Event, Exception, JsonLenientString, Level, Mechanism, Values,
+    ClientSdkInfo, Context, Contexts, Event, Exception, JsonLenientString, Level, Mechanism, Values,
 };
 use relay_protocol::{Annotated, Value};
 
 type Minidump<'a> = minidump::Minidump<'a, &'a [u8]>;
 
+/// Client SDK name used for the event payload to identify events created out of minidumps.
+const CLIENT_SDK_NAME: &str = "minidump";
```
Comment (loewenheim) [inline relay-server/src/utils/native.rs]: I would give this a name to indicate that it applies specifically to minidumps.

**Review by loewenheim — CHANGES_REQUESTED** [review]

**Comment by loewenheim** [comment]
Adding the more specific names sounds like a good idea to me.

**Review by Dav1dde — COMMENTED** [review]
Would be good if you modify existing tests to also assert the client sdk name, we should have at least some tests that can be easily extended to assert the SDK name.

**Review by loewenheim — APPROVED** [review]

File: relay-server/src/utils/native.rs [hunk 2]
```diff
@@ -192,6 +192,23 @@ pub fn process_minidump(event: &mut Event, data: &[u8]) {
...
+        .get_stream::<minidump::MinidumpBreakpadInfo>()
+        .is_ok()
+    {
+        "minidump.breakpad"
+    } else {
+        "minidump.unknown"
+    };
+
+    // Add sdk information for analytics.
+    event.client_sdk.get_or_insert_with(|| ClientSdkInfo {
```
Comment (bruno-garcia) [inline relay-server/src/utils/native.rs:207]: will this override the SDK set in the client? Or this only affects minidumps from the minidump endpoint?

Meaning if it's `sentry-native` attaching a minidump from crashpad, will it no longer send `sentry-native`?

**Pull request merged**